client/src/js/services/ModalService.js

```diff
@@ -69,5 +80,85 @@ function ModalService(Modal) {
     });
   }
 
+  /**
+   * Select cashbox modal
+   */
+  function openSelectCashbox(request) {
+    /**
+     * request contains :
+     * cashboxId => the cashbox id send in url : /cash/:id
+     * cashbox.id => the cashbox id which is in the cache
+     * if no cashbox is set in the cache or in the url display a particular modal
+     */
+    var cashboxIsSet = request.cache_cashbox_id && request.url_cashbox_id;
```

Comment:
In the future, I think we can make the URL take precedence over the cache, so that if a URL cashbox id exists, it is passed in, otherwise, pass in the cache  The pseudocode would look like this:
```js
// if neither exist, this will be null!
var cashboxId = request.url_cashbox_id || request.cache_cashbox_id;
```